lib/osf-components/addon/components/file-browser/add-new/template.hbs

```diff
@@ -0,0 +1,37 @@
+{{#let (unique-id 'upload-button-id') as |uploadButtonId|}}
+    <ResponsiveDropdown
+        @renderInPlace={{true}}
+        @buttonStyling={{true}}
+        @horizontalPosition='left'
+        as |dropdown|
+    >
+        <BsButton
+            local-class='TriggerButton {{if dropdown.isOpen 'CloseButton'}}'
+            @type='success'
+            {{on 'click' dropdown.toggle}}
+        >
+            <FaIcon @icon='plus' @size='2x' @fixedWidth={{true}} />
+        </BsButton>
+        <dropdown.content
+            local-class='DropdownList'
+            @rootEventType='mousedown'
```

Comment:
This is a very long 🐰 trail so you'll have to 🐻 with a rather long explanation:
When we click a button to call up a OS file browser window for selecting files, we are actually using the native HTML file inputs to achieve that functionality. That means there are two vital parts to this functionality:
1. There needs to be an `<input type='file'>` element in the DOM.
2. There needs to be a `click` event triggered on the `<input type='file'>` element. 

Now, one thing that is weird with `dropzone` is that the hidden `<input type='file'>` element would only be inserted into the DOM when the element identified as the trigger (by way of setting the `clickableElementId`) is also in the DOM. This is fine for most cases. But in our particular case, the trigger element is a `<Button>` inside a `<ResponsiveDropdown>`. That means the hidden file input will be inserted when the dropdown is opened, and will be destroyed when the dropdown is closed. To make sure it works, we need to make sure the hidden file input doesn't get destroyed before dropzone is able to process the referenced files into its upload queue in memory.

And this is where the conflict between `dropzone` and `ember-basic-dropdown` happens: When we click the trigger button, because the trigger button itself is NOT the hidden file input, `dropzone` needs to programmatically trigger a `click` event on the hidden file input element. Because of event bubbling, this `click` event will also trigger the `ember-basic-dropdown` `click` event handler attached to the document root. In this event handler, `ember-basic-dropdown` will try to see if the clicked element is within the dropdown or not. If the element is outside the dropdown, it will close the dropdown because it is the expected behavior.

And guess what, the hidden file input element was inserted to the document root element, therefore counted as `outside` of the dropdown. Hence the dropdown is closed, removing the trigger element, in turn destroys the hidden file input element. The whole functionality is in shambles. Now `dropzone` complains about not knowing where the files are because the hidden file input is gone.

By setting `@rootEventType='mousedown'`, we are effectively telling `ember-basic-dropdown`'s event handler at document root to not listen for `click`s but for `mousedown`s instead. This solves the premature closing of dropdown because it no longer binds to the `click` event triggered on the hidden file input while preserving the expected behavior of clicking outside the dropdown, because real mouse clicks will trigger `mousedown`, `mouseup` and `click`.

I hope this explanation is sane enough. I am surely not quite sane now.

Reply:
I wonder if this is going to cause us any troubles on mobile.

Reply:
> I wonder if this is going to cause us any troubles on mobile.

Mobile seems to be fine when testing in mobile mode.

Reply:
Wow, that's quite an interesting problem, and appreciate the explanation as well. And definitely something to look out for in mobile testing too, as there may be some discrepancies in how browsers handle mouse/pointer/touch events. Again, appreciate the deep dive!

Reply:
seems like we *do* control the hidden-file-input lifecycle, because it follows the lifecycle of the trigger element, which follows the lifecycle of the dropdown -- funnily enough, looks like the `onClose` thing is [also in existing code](https://github.com/CenterForOpenScience/ember-osf-web/blob/755a7a0c4aa00e7234094daa047aa1ee0e9d8f02/lib/osf-components/addon/components/files/menu/template.hbs#L3), tho might be `@rootEventType='mousedown'` was added because `isUploading` didn't update in time to fix the problem...

Reply:
...don't let me stand in the way of this PR, tho, i was just pulled in by the long explanation and couldn't help noodling on possibilities 👍

Reply:
> (can control where dropzone inserts its hidden file input by passing `hiddenInputContainer`, but that gets hinky because `DropzoneWidget` is outside the dropdown, unlike the dropzone trigger... ok i'm done now)

I remembered going that route before, but later discovered it's not *where* the hidden file input is inserted that is the problem, cuz now the hidden file input is not inserted to the dropdown, but *when* the hidden file input gets destroyed.